lib/compat/wordpress-6.1/hook-theme-json-for-classic-themes.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+function theme_json_current_theme_has_support( ){
+	return WP_Theme_JSON_Resolver_Gutenberg::theme_has_support();
+}
+
+function theme_json_default_filter_for_classic_themes( $theme_json_data ) {
+	if ( ! theme_json_current_theme_has_support() ) {
+		$new_data = array(
+			'version'  => 2,
+			'settings' => array(
+				/*
+				 * We should be able to remove this.
+				 * Without it, it breaks.
+				 */
+				"spacing" => array(
```

Comment:
The resolver breaks if I create a `WP_Theme_JSON` object without settings. It seems that it's this bit that fails. Haven't been able to look at why but it should be possible to create an object with empty settings, correct? cc @glendaviesnz